lib/OptimizationManopt/src/OptimizationManopt.jl

```diff
@@ -0,0 +1,307 @@
+    :who_knows
+end
+
+## quasi Newton
+
+struct QuasiNewtonOptimizer{Teval <: AbstractEvaluationType,
+    TM <: AbstractManifold, Tretr <: AbstractRetractionMethod,
+    Tvt <: AbstractVectorTransportMethod, TLS <: Stepsize} <:
+       AbstractManoptOptimizer
+    M::TM
+    retraction_method::Tretr
+    vector_transport_method::Tvt
+    stepsize::TLS
+end
+
+function QuasiNewtonOptimizer(M::AbstractManifold;
+        eval::AbstractEvaluationType = InplaceEvaluation(),
+        retraction_method::AbstractRetractionMethod = default_retraction_method(M),
+        vector_transport_method::AbstractVectorTransportMethod = default_vector_transport_method(M),
+        stepsize = WolfePowellLinesearch(M;
+            retraction_method = retraction_method,
+            vector_transport_method = vector_transport_method,
+            linesearch_stopsize = 1e-12))
+    QuasiNewtonOptimizer{typeof(eval), typeof(M), typeof(retraction_method),
+        typeof(vector_transport_method), typeof(stepsize)}(M,
+        retraction_method,
+        vector_transport_method,
+        stepsize)
+end
+
+function call_manopt_optimizer(opt::QuasiNewtonOptimizer{Teval},
+        loss,
+        gradF,
+        x0,
+        stopping_criterion::Union{Nothing, Manopt.StoppingCriterion}) where {
+        Teval <:
+        AbstractEvaluationType
+}
+    sckwarg = stopping_criterion_to_kwarg(stopping_criterion)
+    opts = quasi_Newton(opt.M,
+        loss,
+        gradF,
+        x0;
+        return_state = true,
+        evaluation = Teval(),
+        retraction_method = opt.retraction_method,
+        vector_transport_method = opt.vector_transport_method,
+        stepsize = opt.stepsize,
+        sckwarg...)
+    # we unwrap DebugOptions here
+    minimizer = Manopt.get_solver_result(opts)
+    return (; minimizer = minimizer, minimum = loss(opt.M, minimizer), options = opts),
+    :who_knows
+end
+
+## Optimization.jl stuff
+
+function build_loss(f::OptimizationFunction, prob)
+    function (::AbstractManifold, θ)
+        x = f.f(θ)
+        __x = first(x)
+        return prob.sense === Optimization.MaxSense ? -__x : __x
+    end
+end
+
+function build_gradF(f::OptimizationFunction{true}, prob, cur)
+    function g(M::AbstractManifold, G, θ)
+        f.grad(G, θ, cur...)
+        G .= riemannian_gradient(M, θ, G)
+    end
+    function g(M::AbstractManifold, θ)
+        G = zero(θ)
+        f.grad(G, θ, cur...)
+        return riemannian_gradient(M, θ, G)
+    end
+end
+
+# TODO:
+# 1) convert tolerances and other stopping criteria
+# 2) return convergence information
+# 3) add callbacks to Manopt.jl
+
+function SciMLBase.__solve(prob::OptimizationProblem,
+        opt::AbstractManoptOptimizer,
+        data = Optimization.DEFAULT_DATA;
+        callback = (args...) -> (false),
+        maxiters::Union{Number, Nothing} = nothing,
+        maxtime::Union{Number, Nothing} = nothing,
+        abstol::Union{Number, Nothing} = nothing,
```

Comment:
You can translate `abstol` into `StopWhenChangeLess` I think. `reltol` is really hard to define generically so we don't use that.